File: src/SIL.XForge.Scripture/Models/TextAudio.cs
using System.Collections.Generic;
using SIL.Scripture;
using SIL.XForge.Models;

namespace SIL.XForge.Scripture.Models;

public class TextAudio : ProjectData
{
    public static string GetDocId(string projectId, int book, int chapter) =>
        $"{projectId}:{Canon.BookNumberToId(book)}:{chapter}:target";

    public string DataId { get; set; }
    public List<AudioTiming> Timings { get; set; } = [];
    public string MimeType { get; set; }
    public string AudioUrl { get; set; }
}

File: src/RealtimeServer/scriptureforge/models/text-audio.ts
import { Canon } from '@sillsdev/scripture';
import { PROJECT_DATA_INDEX_PATHS, ProjectData } from '../../common/models/project-data';
import { AudioTiming } from './audio-timing';

export const TEXT_AUDIO_COLLECTION = 'text_audio';
export const TEXT_AUDIO_INDEX_PATHS: string[] = PROJECT_DATA_INDEX_PATHS;

export function getTextAudioId(projectId: string, bookNum: number, chapterNum: number): string {
  return `${projectId}:${Canon.bookNumberToId(bookNum)}:${chapterNum}:target`;
}

export interface TextAudio extends ProjectData {
  dataId: string;
  timings: AudioTiming[];
  mimeType: string;
  audioUrl: string;
}
